### PR TITLE
Minor: Rephrase MSRV policy to be more explanatory

### PR DESCRIPTION
**Pull request by comphead**
## Which issue does this PR close?

Looks like the MSRV policy on hotfixes can be slightly reworded and improved

Closes #.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -131,9 +131,10 @@ Rust versions](https://releases.rs) OR the stable minor Rust version as of 4 mon
 
 For example, given the releases `1.78.0`, `1.79.0`, `1.80.0`, `1.80.1` and `1.81.0` DataFusion will support 1.78.0, which is 3 minor versions prior to the most minor recent `1.81`.
 
-If a hotfix is released for the minimum supported Rust version (MSRV), the MSRV will be the minor version with all hotfixes, even if it surpasses the four-month window.
+Note: If a Rust hotfix is released for the currently supported Minimum Supported Rust Version (MSRV), the MSRV will be updated to the specific minor version that includes all applicable hotfixes. This adjustment will apply even if the release of the hotfix occurs earlier the stipulated four-month support window for a given Rust version.
```
Comment (findepi) [inline README.md:134]: "earlier than"? or maybe actually "later than the start of"?

Comment (comphead) [reply README.md:134]: right, this relative time algebra explodes my brain

Comment (comphead) [inline README.md:134]:
```suggestion
Note: If a Rust hotfix is released for the currently supported Minimum Supported Rust Version (MSRV), the MSRV will be updated to the specific minor version that includes all applicable hotfixes. This adjustment will apply even if the release of the hotfix occurs later than the start of stipulated four-month support window for a given Rust version.
```

Comment (findepi) [reply README.md]: Me too. I can take a stab at proposing some wording

Comment (comphead) [reply README.md]: Just updated, PTAL

**Comment by findepi** [comment]
What about slightly different approach to the wording:

```markdown
## Rust Version Compatibility Policy

The Rust toolchain releases are tracked at [Rust Versions](https://releases.rs) and follow
[semantic versioning](https://semver.org/). A Rust toolchain release can be identified
by a version string like `1.80.0`, or more generally `major.minor.patch`.
A Rust toolchain of given version can be officially released or in preparation,
as stated on the Rust Versions page.

DataFusion's Minimum Required Stable Rust Version (MSRV) policy is the following:

- the four latest stable Rust minor releases are supported.
  For example if the stable releases are `1.81.x`, `1.80.x`, `1.79.x`, `1.78.x`, `1.77.x`,
  then `1.81.x`, `1.80.x`, `1.79.x`, `1.78.x` are the four latest minor releases and are supported.
- the latest stable release as of four months ago is supported.
  For example if the current date is `2024-12-01`, then the latest stable release as of `2024-08-01`
  is supported.

Rust occasionally releases patch releases for a stable minor version.
Only the latest patch release for a stable minor version is supported.

We enforce this policy using a [MSRV CI Check](https://github.com/search?q=repo%3Aapache%2Fdatafusion+rust-version+language%3ATOML+path%3A%2F%5ECargo.toml%2F&type=code)
```

**Comment by comphead** [comment]
> What about slightly different approach to the wording:
> 
> ```
> ## Rust Version Compatibility Policy
> 
> The Rust toolchain releases are tracked at [Rust Versions](https://releases.rs) and follow
> [semantic versioning](https://semver.org/). A Rust toolchain release can be identified
> by a version string like `1.80.0`, or more generally `major.minor.patch`.
> A Rust toolchain of given version can be officially released or in preparation,
> as stated on the Rust Versions page.
> 
> DataFusion's Minimum Required Stable Rust Version (MSRV) policy is the following:
> 
> - the four latest stable Rust minor releases are supported.
>   For example if the stable releases are `1.81.x`, `1.80.x`, `1.79.x`, `1.78.x`, `1.77.x`,
>   then `1.81.x`, `1.80.x`, `1.79.x`, `1.78.x` are the four latest minor releases and are supported.
> - the latest stable release as of four months ago is supported.
>   For example if the current date is `2024-12-01`, then the latest stable release as of `2024-08-01`
>   is supported.
> 
> Rust occasionally releases patch releases for a stable minor version.
> Only the latest patch release for a stable minor version is supported.
> 
> We enforce this policy using a [MSRV CI Check](https://github.com/search?q=repo%3Aapache%2Fdatafusion+rust-version+language%3ATOML+path%3A%2F%5ECargo.toml%2F&type=code)
> ```

I like part about semver so its removes ambiguity about major and minor versions. For the policy itself I feel it can be easier to read... let me reword it

**Comment by comphead** [comment]
@alamb @findepi can I get the review on this?

File: README.md [hunk 2]
```diff
@@ -126,14 +126,18 @@ Optional features:
...
 ## Rust Version Compatibility Policy
 
-DataFusion's Minimum Required Stable Rust Version (MSRV) policy is to support stable [4 latest
-Rust versions](https://releases.rs) OR the stable minor Rust version as of 4 months, whichever is lower.
+The Rust toolchain releases are tracked at [Rust Versions](https://releases.rs) and follow
+[semantic versioning](https://semver.org/). A Rust toolchain release can be identified
+by a version string like `1.80.0`, or more generally `major.minor.patch`.
+
+The Minimum Supported Rust Version (MSRV) policy defines the MSRV as the earliest stable Rust version released either
+within the last four months or from the last four stable [Rust releases](https://releases.rs), whichever has the older release date.
```
Comment (alamb) [inline README.md:134]:
```suggestion
DataFusion's supports the last 4 stable Rust minor versions released and any such versions released within the last 4 months. 
```

**Review by alamb — APPROVED** [review]
Thanks @comphead  - I think this PR is better than what is on main. I have a suggestion to perhaps make it clearer.

**Pull request merged**